packages/chain-adapters/src/utxo/UtxoBaseAdapter.ts

```diff
@@ -219,6 +220,25 @@ export abstract class UtxoBaseAdapter<T extends UtxoChainId> implements IChainAd
 
       const bip44Params = this.getBIP44Params({ accountNumber, accountType, isChange, index })
 
+      if (pubKey) {
+        const account = await this.getAccount(pubKey)
+        const index = bip44Params.isChange
+          ? account.chainSpecific.nextChangeAddressIndex
+          : account.chainSpecific.nextReceiveAddressIndex
+
+        if (index === undefined)
+          throw new Error(`UtxoBaseAdapter: Could not fetch address index from unchained`)
+        if (!account.chainSpecific.addresses)
+          throw new Error(`UtxoBaseAdapter: Could not fetch addresses from unchained`)
+        const address = account.chainSpecific.addresses[index].pubkey
```

Comment:
not sure if this is possible, but just a thought
```suggestion
        const address = account.chainSpecific.addresses[index]?.pubkey
```

Reply:
Absolutely, makes for one less throw just for the sake of type system being happy - a336e4c5f8